onnx/backend/test/case/node/split.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+
+import onnx
+from ..base import Base
+from . import expect
+
+
+class Split(Base):
+
+    @staticmethod
+    def export_1d():
+        input = np.array([1., 2., 3., 4., 5., 6.])
+
+        node = onnx.helper.make_node(
+            'Split',
+            inputs=['input'],
+            outputs=['output_1', 'output_2', 'output_3'],
+            axis=0,
+            split=[2]
```

Comment:
We should not have the split attribute here, because the split length is determined by outputs length.

Reply:
Are you suggesting that the following part from defs.cc is not accurate too? 

"If the 'split' argument contains only one element, and that element divides the length of split axis evenly, the tensor is split into equal sized parts with given length."

Shouldn't it be the case that outputs length is determined by 'split' attribute ?

Reply:
The original doc:
Split a tensor into a list of tensors, along the specified 'axis'. Lengths of the parts can be specified using argument 'split'. Otherwise, the tensor is split to equal sized parts.

If you change the behavior, determine the length according to the split, then it will be a breaking changes. I would like to keep the old behavior. But we need to be more clear on the corner case, i.e., if dim[axis] % length[outputs] != 0.

Reply:
I am not sure why 'split' attributed should not be used in this case even as per the original doc. 'split' attribute is not an optional attribute in original doc and it's defined as:

(From the original doc)
'split' : list of ints
length of each output

So , one interpretation of this definition is that if 'split' is a list only containing one element, that element becomes length of each output.  My proposed change in definition is intended to add more clarity into this interpretation.

Also, a left-to-right logic may define the flow an op more naturally, i.e.,  it may be more natural to use 'input' characteristics  and 'attribute' (in this case 'split') to generate the 'output',  as compared to using 'output' characteristic to define op behavior or possibily interpolate intended 'attribute'.

Reply:
https://github.com/onnx/onnx/blob/master/onnx/defs/tensor/defs.cc#L154 shows that split is an optional attribute, also if it is required, you will find 'required' in the description of this attribute.

The reasons I don't want to use split in this case:
1) It is a breaking change, which requires backend changes...
2) if we use split to specify the length, this is redundant with the length of the output (also static information), which can also help us figure the length.